### DISCUSSION TITLE
Why Prisma force where with JSON_TYPE

### DISCUSSION THREAD

**Discussion by meccin** [prisma/prisma#23561] [Q&A] [answered]
### Question

So, I trying use where in JSON field, but for any reason, Prisma force a second where using JSON_TYPE, but, he's forcing a wrong type. Because, is ARRAY, and he's forcing STRING.

So, my query doesn't work, someone have a little tip to me? How I can force type to ARRAY.

I use:
```
messages: {
    path: '$[*].content',
    string_contains: "search text",
},
```

But, because I put string_contains, he's force type to STRING, but is ARRAY. And Prisma don't have WHERE RAW to use.

Sorry if is on wrong place.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
model Conversation {
  id            String    @id @default(uuid())
  date          DateTime
  messages      Json
  created_at    DateTime  @default(now())
  created_by    String
  updated_at    DateTime  @updatedAt
  updated_by    String
  deleted_at    DateTime?
  deleted_by    String?

  @@map("conversations")
}
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: macOS Sonoma 14.3.1
- Database: MySQL/MariaDB
- Node.js version: v20.11.1

```
prisma                  : 5.10.2
@prisma/client          : 5.10.2
Computed binaryTarget   : darwin-arm64
Operating System        : darwin
Architecture            : arm64
Node.js                 : v20.11.1
Query Engine (Node-API) : libquery-engine 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli 5a9203d0590c951969e85a7d07215503f4672eb9 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.10.0-34.5a9203d0590c951969e85a7d07215503f4672eb9
Default Engines Hash    : 5a9203d0590c951969e85a7d07215503f4672eb9
Studio                  : 0.499.0
```

**Comment by ludralph** [comment]
Hi @avuenja 👋 

Thank you for raising this question.

`string_contains` filter is not applicable when the JSON field is an array. Instead, you should use the [`array_contains`](https://www.prisma.io/docs/orm/reference/prisma-client-reference#array_contains) filter when dealing with arrays. This filter checks for the presence of a specific value in an array. Here's an example:

```js
const getUsers = await prisma.user.findMany({
  where: {
    pets: {
      path: '$.owned.cats',
      array_contains: 'Elliott',
    },
  },
})
```

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by meccin** [reply]
Hello @ludralph 

So, unfortunately doesn't work using `array_contains`, because I'm need search using `LIKE` in a string.

Thinking I have multiples content, inside messages, so I need search using `$[*].content` where return to me `JSON_TYPE == ARRAY`, but I'm search a LIKE using `%my search text%`.

So I need pass `string_contains`, or some way to disable the JSON_TYPE where, that Prisma force in where.

I'll use like it:
```sql
SELECT JSON_EXTRACT('[{"content": "tests"},{"content": "dev"}]', '$[*].content') LIKE "%tests%";
```

But in Prisma is not possible use a raw where, or only pass my simple where, and not put one more where, that I don't want.

Is possible I disable this JSON_TYPE where? Because I don't want it, and only use my JSON_EXTRACT where.

Thank you for answer me, and help me to discover some solution to this problem xD

**Reply by ludralph** [reply, marked as answer]
It is not possible to disable the `JSON_TYPE` where. You can however write [raw query](https://www.prisma.io/docs/orm/prisma-client/queries/raw-database-access/raw-queries#queryraw) in Prisma to achieve the same thing.

```js
const result = await prisma.$queryRaw`SELECT JSON_EXTRACT('[{"content": "tests"},{"content": "dev"}]', '$[*].content') LIKE "%tests%";`
```

**Reply by meccin** [reply]
Hey hey,

Yeah, I think is best solution. Use raw query to do it.

Thank you for help me in this discussion.

Great day.
